### PR TITLE
ngt: 1.8.4 -> 1.12.3-alpha

### PR DESCRIPTION
**Pull request by tomberek**
###### Motivation for this change
Version bump

###### Things done
hash update

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Build errors on darwin:

```
-- The C compiler identification is Clang 7.1.0
-- The CXX compiler identification is Clang 7.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/gkcr1p2x0psdc31bavmqbfb2nwq0d6jn-clang-wrapper-7.1.0/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/gkcr1p2x0psdc31bavmqbfb2nwq0d6jn-clang-wrapper-7.1.0/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- VERSION: 1.12.2
-- CMAKE_BUILD_TYPE: Release
-- CMAKE_BUILD_TYPE_LOWER: release
CMake Error at /nix/store/z7qvx6wbvmjvs767g06h03pkzr5aqn6i-cmake-3.18.2/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
Call Stack (most recent call first):
  /nix/store/z7qvx6wbvmjvs767g06h03pkzr5aqn6i-cmake-3.18.2/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/z7qvx6wbvmjvs767g06h03pkzr5aqn6i-cmake-3.18.2/share/cmake-3.18/Modules/FindOpenMP.cmake:529 (find_package_handle_standard_args)
  CMakeLists.txt:70 (find_package)


-- Configuring incomplete, errors occurred!
See also "/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-NGT-v1.12.2.drv-0/source/build/CMakeFiles/CMakeOutput.log".
See also "/private/var/folders/r4/k85s52jx3555ckj0wty615s00000gp/T/nix-build-NGT-v1.12.2.drv-0/source/build/CMakeFiles/CMakeError.log".
```

**Comment by tomberek** [comment]
@SuperSandro2000 i have some notes with this addtion:
```
buildInputs = [ openmpi ];
NIX_ENFORCE_NO_NATIVE = false;
```
The NIX var may not be needed. Can you test something like that out?

File: pkgs/development/libraries/ngt/default.nix [hunk 1]
```diff
@@ -1,20 +1,24 @@
-{ stdenv, fetchFromGitHub, cmake }:
+{ stdenv, fetchFromGitHub, cmake, openmpi, enableAVX ? true }:
 stdenv.mkDerivation rec {
   pname = "NGT";
-  version = "v1.8.4";
+  version = "v1.12.3-alpha";
   nativeBuildInputs = [ cmake ];
   src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ngt/default.nix:6]:
```suggestion
  nativeBuildInputs = [ cmake ];

  src = fetchFromGitHub {
```

File: pkgs/development/libraries/ngt/default.nix [hunk 2]
```diff
@@ -1,20 +1,24 @@
...
     owner = "yahoojapan";
     repo = "NGT";
-    rev = version;
-    sha256 = "f2019e7916b81f8aeabc57d682904c8447776bf9ba94525d20265b329aa43eb5";
+    rev = "29c88ff6cd5824d3196986d1f50b834565b6c9dd";
+    sha256 = "sha256-nu0MJNpaenOB4+evoSVLKmPIuZXVj1Rm9x53+TfhezY=";
   };
+  buildInputs = [ openmpi ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ngt/default.nix:12]:
```suggestion
  };

  buildInputs = [ llvmPackages.openmp ];
```

File: pkgs/development/libraries/ngt/default.nix [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
+  NIX_ENFORCE_NO_NATIVE=! enableAVX;
+  __AVX2__ = if enableAVX then 1 else 0;
 
   enableParallelBuilding = true;
 
   meta = with stdenv.lib; {
     homepage = "https://github.com/yahoojapan/NGT";
-    description = "Nearest Neighbor Search with Neighborhood Graph and Tree for High-dimensional Data";
+    description =
+      "Nearest Neighbor Search with Neighborhood Graph and Tree for High-dimensional Data";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ngt/default.nix:21]:
```suggestion
    description = "Nearest Neighbor Search with Neighborhood Graph and Tree for High-dimensional Data";
```

File: pkgs/development/libraries/ngt/default.nix [hunk 4]
```diff
@@ -1,20 +1,24 @@
-{ stdenv, fetchFromGitHub, cmake }:
+{ stdenv, fetchFromGitHub, cmake, openmpi, enableAVX ? true }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ngt/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, cmake, llvmPackages, enableAVX ? true }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Still fails with my changes but much further down compiling:

```
7 warnings generated.
[ 87%] Linking CXX executable l2-uint8-range-search
7 warnings generated.
7 warnings generated.
[ 89%] Linking CXX executable ngt
[ 92%] Linking CXX executable cosine-float
[ 92%] Built target l2-uint8-range-search
[ 92%] Built target cosine-float
[ 92%] Built target ngt_exe
make: *** [Makefile:149: all] Error 2
builder for '/nix/store/9ynaf0k66xjk59phgxmn7zf3nhz41nzk-NGT-v1.12.3-alpha.drv' failed with exit code 2
error: build of '/nix/store/9ynaf0k66xjk59phgxmn7zf3nhz41nzk-NGT-v1.12.3-alpha.drv' failed
```

**Comment by tomberek** [comment]
Incorporated your suggestions. I'm getting a hold of a Darwin machine to resolve your build issue.

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 101790` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- ngt
</details>

**Comment by tomberek** [comment]
OSX build works on my end

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101790` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ngt</li>
  </ul>
</details>

**Comment by tomberek** [comment]
@SuperSandro2000 It seems those are AVX instructions failing. Perhaps we've got some hardware differences. Does it work if you remove the AVX?

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 It seems those are AVX instructions failing. Perhaps we've got some hardware differences. Does it work if you remove the AVX?

Yes it builds without avx enabled.

**Comment by tomberek** [comment]
Disabled AVX by default.

File: pkgs/development/libraries/ngt/default.nix [hunk 5]
```diff
@@ -1,15 +1,20 @@
-{ stdenv, fetchFromGitHub, cmake }:
+{ stdenv, fetchFromGitHub, cmake, llvmPackages, enableAVX ? true }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ngt/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, cmake, llvmPackages, enableAVX ? false }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101790` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ngt</li>
  </ul>
</details>

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 101790` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ngt</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101790` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ngt</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101790` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ngt</li>
  </ul>
</details>

**Pull request merged**